pkg/scheduler/framework/types.go

```diff
@@ -39,6 +39,61 @@ import (
 
 var generation int64
 
+// ActionType is an integer to represent one type of resource change.
+// Different ActionTypes can be bit-wised to compose new semantics.
+type ActionType int64
+
+// Constants for ActionTypes.
+const (
+	Add    ActionType = 1 << iota // 1
+	Delete                        // 10
+	// UpdateNodeXYZ is only applicable for Node events.
+	UpdateNodeUnschedulable // 100
+	UpdateNodeAllocatable   // 1000
+	UpdateNodeLabel         // 10000
+	UpdateNodeTaint         // 100000
+	UpdateNodeCondition     // 1000000
+
+	All ActionType = 1<<iota - 1 // 1111111
+
+	// Use the general Update type if you don't either know or care the specific sub-Update type to use.
+	Update = UpdateNodeUnschedulable | UpdateNodeAllocatable |
+		UpdateNodeLabel | UpdateNodeTaint | UpdateNodeCondition
+)
+
+// GVK is short for group/version/kind, which can uniquely represent a particular API resource.
+type GVK string
+
+// Constants for GVKs.
+const (
+	Pod                   GVK = "Pod"
+	Node                  GVK = "Node"
+	PersistentVolume      GVK = "PersistentVolume"
+	PersistentVolumeClaim GVK = "PersistentVolumeClaim"
+	Service               GVK = "Service"
+	StorageClass          GVK = "storage.k8s.io/StorageClass"
+	CSINode               GVK = "storage.k8s.io/CSINode"
+	WildCard              GVK = "*"
+)
+
+// WildCardEvent semantically matches all resources on all actions.
+var WildCardEvent = ClusterEvent{Resource: WildCard, ActionType: All}
+
+// ClusterEvent abstracts how a system resource's state gets changed.
+// Resource represents the standard API resources such as Pod, Node, etc.
+// ActionType denotes the specific change such as Add, Update or Delete.
+type ClusterEvent struct {
+	Resource   GVK
+	ActionType ActionType
+}
+
+// ClusterEventSet embeds two slices to store Interested and NonInterested events.
+// It can represent abundant semantics using least configuration.
```

Comment:
We don't need this.
It's only developers that need to care about this, not even cluster administrators.
Besides, when a new event is added, we would need to update every existing plugin to add that new event to "NotInterested". Most commonly, plugins' interest in events don't change.

Reply:
Although it's for developers, sometimes specifying a full list of events is not a trivial task.

> Besides, when a new event is added, we would need to update every existing plugin to add that new event to "NonInterested".

Developers can achieve this by specifying a wildcard NonIntrested event, and explicitly specify the Interested events.

Note that some (out-of-tree) plugin may not be aware of this new queuing mechanics, and hence still inherits the current "aggressive" queuing mechnics. That implies, those plugins are interested in all events.

Reply:
we can leave the logic as: if nil, then react to everything. But no in-tree plugins should return nil

Reply:
ok, so this answers a comment I made at the beginning about "non-interested". I have the same thought as Aldo, support opt-in for selected or all events, but not opt-out. I would really rather simplify this as much as we can.